core/whistle_number_manager-1.0.0/src/wh_port_request.erl

```diff
@@ -200,3 +203,126 @@ enable_number(N) ->
+        'ok' ->
+            case send_attachements(Url, JObj) of
+                'error' -> 'ok';
+                'ok' -> set_flag(JObj)
+            end
+    end.
+
+-spec send_request(ne_binary(), wh_json:object()) -> 'error' | 'ok'.
+send_request(Url, JObj) ->
+    Id = wh_json:get_value(<<"_id">>, JObj),
+
+    Headers = [{"Content-Type", "application/json"}
+               ,{"User-Agent", wh_util:to_list(erlang:node())}
+              ],
+
+    Uri = binary:bin_to_list(<<Url/binary, "/", Id/binary>>),
+
+    Remove = [
+        <<"_rev">>
+        ,<<"ui_metadata">>
+        ,<<"_attachments">>
+        ,<<"pvt_request_id">>
+        ,<<"pvt_type">>
+        ,<<"pvt_vsn">>
+        ,<<"pvt_account_db">>
+    ],
+    Replace = [
+        {<<"_id">>, <<"id">>}
+        ,{<<"pvt_port_state">>, <<"port_state">>}
+        ,{<<"pvt_account_id">>, <<"account_id">>}
+        ,{<<"pvt_created">>, <<"created">>}
+        ,{<<"pvt_modified">>, <<"modified">>}
+    ],
+    Data = wh_json:encode(wh_json:normalize_jobj(JObj, Remove, Replace)),
+
+    case ibrowse:send_req(Uri, Headers, 'post', Data, []) of
+        {'ok', "200", _Headers, _Resp} ->
+            lager:debug("submitted_port_request successfully sent");
+        _Other ->
+            lager:error("failed to send submitted_port_request ~p", [_Other]),
+            'error'
+    end.
+
+-spec send_attachements(ne_binary(), wh_json:object()) -> 'error' | 'ok'.
+send_attachements(Url, JObj) ->
+    Id = wh_json:get_value(<<"_id">>, JObj),
+    Attachments = wh_json:get_value(<<"_attachments">>, JObj),
+    wh_json:foldl(
+        fun(Key, Value, _) ->
+            case couch_mgr:fetch_attachment(?KZ_PORT_REQUESTS_DB, Id, Key) of
+                {'error', _R} ->
+                    lager:error("failed to fetch attachment ~s : ~p", [Key, _R]),
+                    'error';
+                {'ok', Attachment} ->
+                    send_attachement(Url, Id, Key, Value, Attachment)
+            end
+        end
+        ,'ok'
+        ,Attachments
+    ).
+
+-spec send_attachement(ne_binary(), ne_binary(), ne_binary(), wh_json:object(), any()) -> 'error' | 'ok'.
+send_attachement(Url, Id, Name, Options, Attachment) ->
+    ContentType = wh_json:get_value(<<"content_type">>, Options),
+
+    Headers = [{"Content-Type", binary:bin_to_list(ContentType)}
+               ,{"User-Agent", wh_util:to_list(erlang:node())}
+              ],
+
+    Uri = binary:bin_to_list(<<Url/binary, "/", Id/binary, "/", Name/binary>>),
+
+    case ibrowse:send_req(Uri, Headers, 'post', Attachment, []) of
+        {'ok', "200", _Headers, _Resp} ->
+            lager:debug("attachment ~s for submitted_port_request successfully sent", [Name]);
+        _Other ->
+            lager:error("failed to send attachment ~s for submitted_port_request ~p", [Name, _Other]),
+            'error'
+    end.
+
+-spec set_flag(wh_json:object()) -> 'ok'.
+set_flag(JObj) ->
+    Doc = wh_json:set_value(<<"sent">>, 'true', JObj),
```

Comment:
I think it should still be a pvt_ field that is then leaked. We don't want the UI setting "sent" on the port request document and expecting it to be valid.